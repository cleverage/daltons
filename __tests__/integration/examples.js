const fs = require('fs')
const path = require('path')

const examplesPath = path.join(__dirname, '../../examples')

describe('examples', () => {
  const SUT = require('../../src/index.js')

  // Declare tests automatically depending on the existing folders within ./examples
  // Use sync method in order to declare test cases synchronously
  const examples = fs.readdirSync(examplesPath)

  examples.forEach(exampleName => {
    it(
      `${exampleName} example should return the list of perfect width for its image and contexts`,
      async () => {
        expect.assertions(1)

        const configurationFile = path.join(
          examplesPath,
          `${exampleName}/configuration.js`,
        )
        const resultsFile = path.join(examplesPath, `${exampleName}/result.js`)

        const config = require(configurationFile)
        const actualResults = await SUT(config)

        expect(actualResults).toMatchSnapshot()
      },
      600000,
    )
  })
})
