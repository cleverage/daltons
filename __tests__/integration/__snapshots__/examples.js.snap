// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`examples nicolas-hoizey.com example should return the list of perfect width for its image and contexts 1`] = `
Map {
  800 => 0.5477375936969963,
  1600 => 0.23050828910652732,
  676 => 0.039722055041855886,
  972 => 0.03271871751381518,
  1000 => 0.03129616457843191,
  974 => 0.0166876402035345,
  648 => 0.014061388630519232,
  576 => 0.012091699950757782,
  1200 => 0.012091699950757782,
  1014 => 0.008973026207802156,
  1119 => 0.008808885484488702,
  1296 => 0.005143075997154894,
  720 => 0.00415823165727417,
  1382 => 0.0034469551895825352,
  1299 => 0.003009246594079991,
  2000 => 0.002626251573015265,
  486 => 0.0017508343820101767,
  1400 => 0.0014225529353832686,
  432 => 0.0011489850631941786,
  975 => 0.0008207036165672703,
  881 => 0.0008207036165672703,
  1440 => 0.0007659900421294523,
  1080 => 0.0007112764676916343,
  717 => 0.0006565628932538163,
  873 => 0.0006018493188159982,
  1502 => 0.0005471357443781802,
  528 => 0.0005471357443781802,
  650 => 0.0004924221699403622,
  1800 => 0.0004924221699403622,
  640 => 0.0004924221699403622,
  514 => 0.00038299502106472615,
  960 => 0.00038299502106472615,
  973 => 0.00038299502106472615,
  370 => 0.00032828144662690813,
  600 => 0.00032828144662690813,
  1496 => 0.00032828144662690813,
  1512 => 0.00032828144662690813,
  2400 => 0.00032828144662690813,
  1040 => 0.00032828144662690813,
  1120 => 0.0002735678721890901,
  1510 => 0.0002735678721890901,
  884 => 0.0002735678721890901,
  1152 => 0.0002735678721890901,
  1297 => 0.00021885429775127209,
  1455 => 0.00021885429775127209,
  840 => 0.00021885429775127209,
  245 => 0.00021885429775127209,
  1556 => 0.00021885429775127209,
  718 => 0.00016414072331345406,
  743 => 0.00016414072331345406,
  768 => 0.00016414072331345406,
  1100 => 0.00016414072331345406,
  1301 => 0.00016414072331345406,
  721 => 0.00016414072331345406,
  613 => 0.00016414072331345406,
  1092 => 0.00016414072331345406,
  760 => 0.00016414072331345406,
  1761 => 0.00016414072331345406,
  1540 => 0.00016414072331345406,
  1829 => 0.00016414072331345406,
  1360 => 0.00016414072331345406,
  1408 => 0.00016414072331345406,
  751 => 0.00016414072331345406,
  977 => 0.00010942714887563604,
  924 => 0.00010942714887563604,
  235 => 0.00010942714887563604,
  1443 => 0.00010942714887563604,
  1074 => 0.00010942714887563604,
  560 => 0.00010942714887563604,
  1320 => 0.00010942714887563604,
  684 => 0.00010942714887563604,
  1374 => 0.00010942714887563604,
  863 => 0.00010942714887563604,
  700 => 0.00010942714887563604,
  1414 => 0.00010942714887563604,
  729 => 0.00010942714887563604,
  761 => 0.00010942714887563604,
  775 => 0.00010942714887563604,
  828 => 0.00010942714887563604,
  1576 => 0.00010942714887563604,
  936 => 0.00010942714887563604,
  1778 => 0.00010942714887563604,
  1044 => 0.00010942714887563604,
  1117 => 0.00010942714887563604,
  1116 => 0.00010942714887563604,
  1067 => 0.00010942714887563604,
  1251 => 0.00010942714887563604,
  1412 => 0.00010942714887563604,
  846 => 0.00010942714887563604,
  680 => 0.00010942714887563604,
  540 => 0.00010942714887563604,
  1148 => 0.00010942714887563604,
  480 => 0.00010942714887563604,
  770 => 0.00010942714887563604,
  1005 => 0.00005471357443781802,
  2100 => 0.00005471357443781802,
  1280 => 0.00005471357443781802,
  1920 => 0.00005471357443781802,
  1144 => 0.00005471357443781802,
  1298 => 0.00005471357443781802,
  433 => 0.00005471357443781802,
  971 => 0.00005471357443781802,
  360 => 0.00005471357443781802,
  371 => 0.00005471357443781802,
  1182 => 0.00005471357443781802,
  436 => 0.00005471357443781802,
  471 => 0.00005471357443781802,
  1006 => 0.00005471357443781802,
  506 => 0.00005471357443781802,
  1036 => 0.00005471357443781802,
  523 => 0.00005471357443781802,
  1066 => 0.00005471357443781802,
  1078 => 0.00005471357443781802,
  1082 => 0.00005471357443781802,
  1106 => 0.00005471357443781802,
  1520 => 0.00005471357443781802,
  1212 => 0.00005471357443781802,
  1246 => 0.00005471357443781802,
  1250 => 0.00005471357443781802,
  1231 => 0.00005471357443781802,
  629 => 0.00005471357443781802,
  1262 => 0.00005471357443781802,
  1288 => 0.00005471357443781802,
  1290 => 0.00005471357443781802,
  1302 => 0.00005471357443781802,
  655 => 0.00005471357443781802,
  658 => 0.00005471357443781802,
  1730 => 0.00005471357443781802,
  1986 => 0.00005471357443781802,
  1338 => 0.00005471357443781802,
  671 => 0.00005471357443781802,
  677 => 0.00005471357443781802,
  682 => 0.00005471357443781802,
  685 => 0.00005471357443781802,
  691 => 0.00005471357443781802,
  864 => 0.00005471357443781802,
  692 => 0.00005471357443781802,
  694 => 0.00005471357443781802,
  1402 => 0.00005471357443781802,
  706 => 0.00005471357443781802,
  710 => 0.00005471357443781802,
  714 => 0.00005471357443781802,
  715 => 0.00005471357443781802,
  719 => 0.00005471357443781802,
  722 => 0.00005471357443781802,
  1468 => 0.00005471357443781802,
  734 => 0.00005471357443781802,
  736 => 0.00005471357443781802,
  738 => 0.00005471357443781802,
  740 => 0.00005471357443781802,
  741 => 0.00005471357443781802,
  1853 => 0.00005471357443781802,
  752 => 0.00005471357443781802,
  754 => 0.00005471357443781802,
  1508 => 0.00005471357443781802,
  756 => 0.00005471357443781802,
  1151 => 0.00005471357443781802,
  900 => 0.00005471357443781802,
  1542 => 0.00005471357443781802,
  1546 => 0.00005471357443781802,
  1548 => 0.00005471357443781802,
  1552 => 0.00005471357443781802,
  1554 => 0.00005471357443781802,
  1558 => 0.00005471357443781802,
  780 => 0.00005471357443781802,
  782 => 0.00005471357443781802,
  786 => 0.00005471357443781802,
  789 => 0.00005471357443781802,
  1584 => 0.00005471357443781802,
  1586 => 0.00005471357443781802,
  797 => 0.00005471357443781802,
  798 => 0.00005471357443781802,
  1598 => 0.00005471357443781802,
  1065 => 0.00005471357443781802,
  3200 => 0.00005471357443781802,
  1256 => 0.00005471357443781802,
}
`;

exports[`examples simple example should return the list of perfect width for its image and contexts 1`] = `
Map {
  511 => 0.20306513409961685,
  510 => 0.13409961685823754,
  643 => 0.12260536398467432,
  513 => 0.10727969348659004,
  770 => 0.10344827586206896,
  512 => 0.08812260536398467,
  567 => 0.08812260536398467,
  515 => 0.06896551724137931,
  1024 => 0.04597701149425287,
  514 => 0.02681992337164751,
  1022 => 0.011494252873563218,
}
`;
